src/aks-preview/azext_aks_preview/aks_app_commands/aksappcommands.py

```diff
@@ -0,0 +1,121 @@
+    # Filename depends on the operating system
+    filename = "draftv2-" + operatingSystem.lower() + "-amd64"
+
+    if not paths:
+        print("List of given DraftV2 paths is empty")
+        return None
+
+    for path in paths:
+        binaryFilePath = path + "/" + filename
+        if os.path.exists(binaryFilePath):
+            print("Existing binary found at: " + binaryFilePath)
+            return binaryFilePath
+    return ""
+
+
+# Returns a list of potential draftV2 binary paths
+def get_potential_paths() -> List[str]:
+    result = [str(Path.home()) + "/" + ".aksapp"]
+    paths = os.environ['PATH'].split(':')
+
+    for path in paths:
+        if "draftv2" in path:
+            result.append(path)
+    return result
+
+
+def run_binary(path: str):
+    if path is None:
+        raise ValueError("The given Binary path was null or empty")
+
+    print("Running DraftV2 Binary ...")
+    process = subprocess.Popen([path], stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    exit_code = process.wait()
+    print(stdout, stderr, exit_code)
+    return exit_code
+
+
+def cmd_finish():
+    # Clean up logic can go here if needed
+    print("We are done Stop.")
+
+
+def download_binary() -> str:
+    # prompt user to download binary. If users says no, we error out and tell them that this requires the binary
+    from knack.prompting import prompt_y_n
+    msg = 'he required binary was not found. Would you like us to download the required binary for you?'
+
+    if not prompt_y_n(msg, default="n"):
+        raise ValueError("`az aks app` requires the missing dependency")
+
+    print("Attempting to download dependency...")
+
+    operatingSystem = platform.system()
+    draftV2ReleaseVersion = "v0.0.5"
+    filename = "draftv2-" + operatingSystem.lower() + "-amd64"
+    url = "https://github.com/Azure/aks-app/releases/download/" + draftV2ReleaseVersion + "/" +  filename
+    headers = {'Accept': 'application/octet-stream'}
+
+    # Downloading the file by sending the request to the URL
+    req = requests.get(url, headers=headers)
+    binaryPath = str(Path.home()) + "/" +".aksapp"
+
+    # Directory
+    if os.path.exists(binaryPath) is False:
+        os.chdir(str(Path.home()))
+        os.mkdir(".aksapp")
+        print("Directory '% s' was created inside of your HOME directory" % ".aksapp")
+
+    if req.ok:
+        # Split URL to get the file name "draftv2-darwin-amd64"
+        os.chdir(binaryPath)
+        # Writing the file to the local file system
+        with open(filename, 'wb') as output_file:
+            output_file.write(req.content)
+        print("Download of DraftV2 binary was successful with a status code: " + str(req.status_code))
+        os.chmod(binaryPath + "/" + filename, 0o777)
+        return binaryPath + "/" + filename
+
+    print("Download of DraftV2 binary was unsuccessful with a status code: " + str(req.status_code))
+    return None
```

Comment:
📓 for this error: `121:1: W391 blank line at end of file` and here is the flake8 rule which emphasise on it: https://www.flake8rules.com/rules/W391.html 

solution: add one blank line in the end of file.

Reply:
Sorry, all of the whitespace, blank line stuff should have been fixed. Maybe I did not push the commit or something. I was only stumped on the type annotation stuff. Thanks